smithy-build/src/main/java/software/amazon/smithy/build/SmithyBuildImpl.java

```diff
@@ -283,13 +283,24 @@ private ProjectionResult applyProjection(
                 .model(projectedModel)
                 .events(modelResult.getValidationEvents());
 
+        // Keep track of the first error created by plugins to fail the build after all plugins have run
+        Throwable firstPluginError = null;
         for (Map.Entry<String, ObjectNode> entry : resolvePlugins(projection).entrySet()) {
             if (pluginFilter.test(entry.getKey())) {
-                applyPlugin(projectionName, projection, baseProjectionDir, entry.getKey(), entry.getValue(),
+                try {
+                    applyPlugin(projectionName, projection, baseProjectionDir, entry.getKey(), entry.getValue(),
                             projectedModel, resolvedModel, modelResult, resultBuilder);
+                } catch (Throwable e) {
+                    LOGGER.severe(String.format("Plugin `%s` failed: %s", entry.getKey(), e));
```

Comment:
I think this would output the first error twice since we're still throwing it _and_ logging it now. Would be good to see how this ends up looking in practice. If it's not good, then... don't log the first error?

Reply:
Yea, it logs every plugin failure, but only throws the first one. [This comment](https://github.com/awslabs/smithy/pull/1416#discussion_r981497343) on the original PR explains why I made that choice, but in short it is to show all plugin failures instead of just the first one. Here's a comparison of what it looks like:
Current state
```
INFO: Applying `broken` plugin to `source` projection

1 Smithy build projections failed.

(source): java.lang.RuntimeException: broken

software.amazon.smithy.build.SmithyBuildException: 1 Smithy build projections failed.

(source): java.lang.RuntimeException: broken
```

Logging plugin errors (what this PR does)
```
INFO: Applying `broken` plugin to `source` projection
SEVERE: Plugin `broken` failed: java.lang.RuntimeException: broken
INFO: Applying `build-info` plugin to `source` projection
INFO: Applying `model` plugin to `source` projection
INFO: Applying `sources` plugin to `source` projection
INFO: Writing empty `source` manifest because no Smithy sources found
INFO: Applying `test1Serial` plugin to `source` projection

1 Smithy build projections failed.

(source): java.lang.RuntimeException: broken

software.amazon.smithy.build.SmithyBuildException: 1 Smithy build projections failed.

(source): java.lang.RuntimeException: broken
```

Not logging the plugin errors
```
INFO: Applying `broken` plugin to `source` projection
INFO: Applying `build-info` plugin to `source` projection
INFO: Applying `model` plugin to `source` projection
INFO: Applying `sources` plugin to `source` projection
INFO: Writing empty `source` manifest because no Smithy sources found
INFO: Applying `test1Serial` plugin to `source` projection

1 Smithy build projections failed.

(source): java.lang.RuntimeException: broken

software.amazon.smithy.build.SmithyBuildException: 1 Smithy build projections failed.

(source): java.lang.RuntimeException: broken
```

Like I mentioned in the comment on the original commit, I think it would be better if we aggregated all plugin failures into one exception.

Reply:
I think @mtdowling is asking to _not_ log this if the `firstPluginError` is `null`, since that one will end up being thrown.

Reply:
I've opened #1762 with those changes. Looking at the output now, I think its nicer to not log the first error since it doesn't do that right now.